msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -555,51 +561,104 @@ public void run() {
     }
 
     /**
-     * Returns the IAccount object matching the supplied home_account_id.
+     * Returns the IAccount object matching the supplied object id.
      *
-     * @param homeAccountIdentifier The home_account_id of the sought IAccount.
-     * @param authority             The authority of the sought IAccount.
-     * @return The IAccount stored in the cache or null, if no such matching entry exists.
+     * @param oid The object id of the sought IAccount.
+     * @param callback AccountLoadedCallback
      */
-    @Nullable
-    public IAccount getAccount(@NonNull final String homeAccountIdentifier,
-                               @Nullable final String authority) {
+    public void getAccount(@NonNull final String oid,
```

Comment:
I believe that we'll need to support getting an account by OID, HomeAccountId, etc... so let's make that possible by accepting a parameter object.